src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -340,6 +355,19 @@ public void RemoveThrottledChecks(ICheckContext checkContext)
             }
         }
 
+        public void RemoveChecks(List<CheckWrapper> checkToRemove, ICheckContext checkContext)
+        {
+            foreach (CheckWrapper check in checkToRemove)
+            {
+                var checkFactory = _checkRegistry.Find(c => c.MaterializedCheck == check);
+                if (checkFactory is not null)
+                {
+                    checkContext.DispatchAsCommentFromText(MessageImportance.High, $"Dismounting check '{check.Check.FriendlyName}'");
+                    RemoveCheck(checkFactory);
+                }
+            }
+        }
```

Comment:
`RemoveChecks` and `RemoveThrottledChecks` both are removing checks that are matching some condition - can we unify those?